### PR TITLE
Dont timeout servre

### PR DESCRIPTION
**Pull request by merituulie**
- CGI timeout exception added
- Added software name and server name to cgi env vars
- Added a waitpid call to after kill signal
- removed connection timeout checks etc
- remove pointers to servers in serverhandler
- dont timeout the poll, just do another call

### CODE REVIEW

File: src/ServerHandler.cpp [hunk 1]
```diff
@@ -15,7 +15,11 @@ ServerHandler::~ServerHandler()
 	}
 
 	if (!_servers.empty())
+	{
+		for (auto serv : _servers)
```
Comment (forbidden-arts) [inline src/ServerHandler.cpp:19]: you're using auto. my head just exploded.

File: src/ServerHandler.cpp [hunk 2]
```diff
@@ -348,17 +318,15 @@ void ServerHandler::runServers(std::map<std::string, Server*> &servers)
 		removeTimedOutClientsAndConnections();
 		// Wait max 3 minutes for incoming traffic
 		int result = poll(_pollfds.data(), _pollfds.size(), CONNECTION_TIMEOUT);
-		if (result == 0)
-		{
-			closeConnections();
-			throw TimeOutException("The program excited with timeout");
-		}
-		else if (result < 0)
+		if (result < 0)
```
Comment (forbidden-arts) [inline src/ServerHandler.cpp:321]: is there a case where it equals zero?

File: src/ServerHandler.cpp (hunk 1 above)
Comment (merituulie) [reply src/ServerHandler.cpp:19]: I know Im getting tired of carrying this group

Comment (merituulie) [reply src/ServerHandler.cpp:19]: 😂

File: src/ServerHandler.cpp (hunk 2 above)
Comment (merituulie) [reply src/ServerHandler.cpp:321]: nope, we dont want to quit even if we dont get any traffic

Comment (merituulie) [reply src/ServerHandler.cpp:321]: result = 0 means we just didnt get any traffic in or out, so lets handler requests we already have and see then if there is more requests coming in.

**Review by forbidden-arts — APPROVED** [review]

**Review by rosamakinen — APPROVED** [review]

**Pull request merged**